fdbserver/masterserver.actor.cpp

```diff
@@ -1241,14 +1234,20 @@ ACTOR Future<Void> serveLiveCommittedVersion(Reference<MasterData> self) {
 				reply.locked = self->databaseLocked;
 				reply.metadataVersion = self->proxyMetadataVersion;
 				reply.minKnownCommittedVersion = self->minKnownCommittedVersion;
-				std::size_t count = 0;
-				if (self->percentageOfTagsToBeSent > 100) {
-					count = rand() % (self->allTags.size() + 1);
-				} else {
-					count = (self->percentageOfTagsToBeSent * self->allTags.size()) / 100;
+
+				int tagCount = 0;
+				if (SERVER_KNOBS->NUMBER_OF_SS_TAGS_TO_SEND > 0) {
+					tagCount = std::min(SERVER_KNOBS->NUMBER_OF_SS_TAGS_TO_SEND, (int)self->allTags.size());
+				} else if (SERVER_KNOBS->PERCENTAGE_OF_SS_TAGS_TO_SEND > 0) {
+					if (SERVER_KNOBS->PERCENTAGE_OF_SS_TAGS_TO_SEND > 110){
+					tagCount = deterministicRandom()->randomInt(0, self->allTags.size() + 1);
+					} else {
+						tagCount = (SERVER_KNOBS->PERCENTAGE_OF_SS_TAGS_TO_SEND * self->allTags.size()) / 100;
+					}
 				}
-				for (std::size_t i=0; i<count; i++) {
-					reply.tags.emplace_back(self->allTags[i]);
+				ASSERT(tagCount <= self->allTags.size());
```

Comment:
Test "rare/ConflictRangeCheck.toml" failed on this assert.  Here are the values that caused this assertion failure:
tagCount = 17, "seld->allTags.size()" = 16. How is it possible for "tagCount" to have value 17 here? Am I not invoking "deterministicRandom()->randomInt()" properly? Any clues? Thanks!

Reply:
```
tagCount = (SERVER_KNOBS->PERCENTAGE_OF_SS_TAGS_TO_SEND * self->allTags.size()) / 100;
```
so if `PERCENTAGE_OF_SS_TAGS_TO_SEND` is `110` this will be execute (IIUC). `110*16/100 = 17`.

Reply:
Thank you so much!